shared/AppInsightsCommon/src/ThrottleMgr.ts

```diff
@@ -0,0 +1,186 @@
+import {
+    IAppInsightsCore, IDiagnosticLogger, _eInternalMessageId, _throwInternal, eLoggingSeverity, isNullOrUndefined, safeGetLogger, strTrim
+} from "@microsoft/applicationinsights-core-js";
+import { IThrottleMsgKey } from "./Enums";
+import { IthrottleDate, IthrottleLocalStorageObj, IthrottleMgrConfig, IthrottleResult } from "./Interfaces/IThrottleMgr";
+import { utlCanUseLocalStorage, utlGetLocalStorage, utlSetLocalStorage } from "./StorageHelperFuncs";
+
+const THROTTLE_STORAGE_PREFIX = "appInsightsThrottle";
+
+
+export class ThrottleMgr {
```

Comment:
Tests showing expected usage would be good and helpful in validating and following the flow.